pkg/api/vlabs/validate.go

```diff
@@ -470,6 +466,38 @@ func (a *Properties) validateAgentPoolProfiles(isUpdate bool) error {
 	return nil
 }
 
+func (a *Properties) validateZones() error {
+	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
+		// all zones or no zones should be defined for the cluster
+		if a.HasAvailabilityZones() {
+			if a.IsClusterAllAvailabilityZones() {
+				// master profile
+				if a.MasterProfile.AvailabilityProfile != VirtualMachineScaleSets {
+					return errors.New("Availability Zones are not supported with an AvailabilitySet. Please set availabilityProfile to VirtualMachineScaleSets")
+				}
+				if a.MasterProfile.Count < len(a.MasterProfile.AvailabilityZones)*2 {
+					return errors.New("the node count and the number of availability zones provided can result in zone imbalance. To achieve zone balance, each zone should have at least 2 nodes or more")
+				}
+				// agent pool profiles
+				for _, agentPoolProfile := range a.AgentPoolProfiles {
+					if agentPoolProfile.AvailabilityProfile == AvailabilitySet {
+						return errors.New("Availability Zones are not supported with an AvailabilitySet. Please either remove availabilityProfile or set availabilityProfile to VirtualMachineScaleSets")
+					}
+					if agentPoolProfile.Count < len(agentPoolProfile.AvailabilityZones)*2 {
+						return errors.New("the node count and the number of availability zones provided can result in zone imbalance. To achieve zone balance, each zone should have at least 2 nodes or more")
+					}
+				}
+				if a.OrchestratorProfile.KubernetesConfig != nil && a.OrchestratorProfile.KubernetesConfig.LoadBalancerSku != "Standard" {
```

Comment:
`&& a.OrchestratorProfile.KubernetesConfig.LoadBalancerSku != nil` 


Defined KubernetesConfig but no LoadBalancerSku defined should not fail